tests/unittest/test_batchify.py

```diff
@@ -37,62 +40,69 @@ def _verify_padded_arr(padded_arr, original_arr, pad_axis, pad_val, pad_length,
-                            _verify_padded_arr(batch_data[i], random_data_npy[i], axis, pad_val, pad_length, dtype)
-                        # Each instance contains 3 arrays, we pad part of them according to index
-                        TOTAL_ELE_NUM = 3
-                        for pad_index in [[0], [1], [2], [0, 1], [1, 2], [0, 1, 2]]:
-                            shapes = [[[2 for _ in range(ndim)] for _ in range(batch_size)]
-                                      for _ in range(TOTAL_ELE_NUM)]
-                            for j in pad_index:
-                                for i in range(batch_size):
-                                    shapes[j][i][axis] = np.random.randint(length_min, length_max)
-                            random_data_npy = [tuple(np.random.normal(0, 1, shapes[j][i]).astype(dtype)
-                                                     for j in range(TOTAL_ELE_NUM)) for i in range(batch_size)]
-                            batchify_fn = []
-                            for j in range(TOTAL_ELE_NUM):
-                                if j in pad_index:
-                                    batchify_fn.append(batchify.Pad(axis=axis, pad_val=pad_val, ret_length=True))
-                                else:
-                                    batchify_fn.append(batchify.Stack())
-                            batchify_fn = batchify.Tuple(batchify_fn)
-                            ret_use_npy = batchify_fn(random_data_npy)
-                            ret_use_mx = batchify_fn([tuple(mx.nd.array(ele[i], dtype=dtype) for i in range(TOTAL_ELE_NUM)) for ele in random_data_npy])
-                            for i in range(TOTAL_ELE_NUM):
-                                if i in pad_index:
-                                    assert ret_use_npy[i][0].dtype == ret_use_mx[i][0].dtype == dtype
-                                    assert ret_use_npy[i][1].dtype == ret_use_mx[i][1].dtype == np.int32
-                                    assert_allclose(ret_use_npy[i][0].asnumpy(),
-                                                    ret_use_mx[i][0].asnumpy())
-                                    assert_allclose(ret_use_npy[i][1].asnumpy(),
-                                                    ret_use_mx[i][1].asnumpy())
-                                    assert(ret_use_npy[i][1].shape == (batch_size,))
-                                else:
-                                    assert ret_use_npy[i].dtype == ret_use_mx[i].dtype == dtype
-                                    assert_allclose(ret_use_npy[i].asnumpy(), ret_use_mx[i].asnumpy())
+                        for _dtype in [None, dtype]:
+                            shapes = [[2 for _ in range(ndim)] for _ in range(batch_size)]
+                            for i in range(len(shapes)):
+                                shapes[i][axis] = np.random.randint(length_min, length_max)
+                            random_data_npy = [np.random.normal(0, 1, shape).astype(dtype)
```

Comment:
Since we set the random_data_npy to have `dtype`, the check also does not test the case that the `dtypes` are different.